benchmarks/utilities/log_parser.hpp

```diff
@@ -151,7 +151,7 @@ inline std::vector<event> parse_csv(std::string const& filename)
 
   auto parse_pointer = [](std::string const& str, uintptr_t& ptr) {
     auto const base{16};
-    ptr = std::stoll(str, nullptr, base);
+    ptr = (str == "(nil)") ? 0 : std::stoll(str, nullptr, base);
```

Comment:
My main question here is that since `%p` is documented as producing implementation-defined behavior and I haven't tested this on enough different systems, should we be concerned about the potential for differences? Given that CI is passing here I'm comfortable saying no, but I wanted to get your opinion. We could also put together a small snippet on Godbolt to see what it outputs on different platforms.

Reply:
Clang and GCC both print "(nil)" on Godbolt. MSVC prints "00000000". 

The nice thing about `%p` is that for non-null pointers you get the `0x` prefix that `%llx` doesn't have: 
code: 
```
int x = 10;
printf("p null: %p, non-null %p\n", nullptr, &x);
printf("llx null: %llx, non-null %llx\n", nullptr, &x);
```
output: 
```
p null: (nil), non-null 0x7ffdc74cc08c
llx null: 0, non-null 7ffdc74cc08c
```

Note that either way I think we should leave this change to `parse_pointer` in place to avoid the runtime errors caused by calling `stoll` on "(nil)".

Reply:
I preferred %p formatting when I was testing in #1722 for the hex prefix as well, I was mostly concerned about whether `nil` was a reliable output. It seems like it's standard enough as long as clang and gcc both do it. I'm not concerned about MSVC. So I say let's keep %p, then.

Reply:
I agree.